lab_2_keywords_cooccurrence/main.py

```diff
@@ -30,13 +29,28 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
 
     In case of corrupt input arguments, None is returned
     """
-    if not type_check(text, str):
+
+    if not isinstance(text, str) or len(text) == 0:
         return None
-    expression = re.compile(r"(?<=^)[^\s\w]+"  # punctuation after the line beginning
-                            r"|(?<=\s)[^\s\w]+"  # punctuation after the whitespace symbol
-                            r"|[^\s\w]+(?=\s)"  # punctuation before the whitespace symbol
-                            r"|[^\s\w]+(?=$)")  # punctuation before the end of line
-    return [clean for phrase in re.split(expression, text) if (clean := phrase.strip())]
+
+    punct = '''.,;:¡!¿?…⋯‹›«»\\"“”\[\]()⟨⟩}{&]|[-–~—]'''
+    for symbol in punct:
+        text = text.replace(symbol, ',')
+        # проходится по всем знакам пункуации и заменяет все на запятые
+    coma_split = text.split(',')
+    # разделяет по запятым на элементы, далее работа со списком
+    final_list = []
+    for i in coma_split:
```

Comment:
you can write it in a one line with list comprehension. But also, I believe you have to call the `lower` method of the string